tools/scripts/build_projects.py

```diff
@@ -81,6 +81,25 @@ def shell_source(script):
 
 	os.environ.update(env)
 
+def re_run_stm32(cmd):
+	global ERR
+	log("Project first failed, rebuild to check if the error persists")
+	log("make reset")
+	cmd_reset = cmd.replace("all", "reset")
+	err = os.system(cmd_reset + ' >> %s 2>&1' % log_file)
+	if err != 0:
+		ERR = 1
+		log_err("ERROR")
+		return err
+
+	log("make all")
+	err = os.system(cmd + ' >> %s 2>&1' % log_file)
+	if err != 0:
+		log("Error persits, not a random fail, please check!")
```

Comment:
I mean, chances are small but the project can fail a second time randomly. I am not sure these logs are that useful. I would simply return err. If the project fails the second time you will have to check the error logs anyways to see what kind of error ocurred. 
if the project does not fail a second time you will just see the successful build and you will not check the logs because the build was successful.

Reply:
If the project is failing 2 consecutive times, we just consider that that's not a random failure anymore.